File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) {
    return 0;
}
        
        int n = nums.size();
        int unique_index = 0;
        
        for(int i = 1; i<n; ++i){
            if(nums[i] != nums[unique_index]){
                 unique_index += 1;
                nums[unique_index] = nums[i];
            }
               
        }
            
        
        return unique_index + 1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        
        n = len(nums)
        unique_index = 0  # Index of the last unique element
        
        for i in range(1, n):
            if nums[i] != nums[unique_index]:
                unique_index += 1
                nums[unique_index] = nums[i]
        
        return unique_index + 1